src/storage/btree_load.c

```diff
@@ -4909,23 +4870,26 @@ online_index_builder (THREAD_ENTRY * thread_p, BTID_INT * btid_int, HFID * hfids
       cur_record.data = NULL;
 
       sc = heap_next (thread_p, &hfids[cur_class], &class_oids[cur_class], &cur_oid, &cur_record, scancache, COPY);
-      if (sc == S_ERROR)
-	{
-	  ASSERT_ERROR_AND_SET (ret);
-	  break;
-	}
-      else if (sc == S_END)
-	{
-	  break;
-	}
+      if (sc != S_SUCCESS)
```

Comment:
It is trivial.
Is it to reduce comparison in case of S_SUCCESS?

Reply:
Yes, to reduce comparisons of if statement.